src/hotspot/share/opto/compile.cpp

```diff
@@ -4815,29 +4815,23 @@ void Compile::sort_macro_nodes() {
 }
 
 void Compile::print_method(CompilerPhaseType cpt, int level) {
-  print_method_impl(cpt, NOT_PRODUCT(CompilerPhaseTypeHelper::to_string(cpt) COMMA) level);
+  print_method(cpt, NULL, level);
 }
 
 void Compile::print_method(CompilerPhaseType cpt, Node* n, int level) {
```

Comment:
Could we also turn `Node* n` into a parameter with default argument `NULL`? Then we can merge both `print_method` methods into one.

Reply:
yes, it's possible. all we need to do is to flip the order from "Node* n, int level" to "int level, Node* n = NULL"